analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -1163,7 +1163,7 @@ def extend_compilation_database_entries(compilation_database):
             source_files = []
             source_dir = entry['directory']
 
-            options = shlex.split(entry['command'])
+            options = shlex.split(entry['command'], posix=(os.name=='posix'))
```

Comment:
The `posix=` parameter is designed to manage nuances between different Unix shells, so it's very mysterious how this helps with Windows and why?!

I realize this is still just a draft, but some... draft comments and a commit message draft would not hurt.

Generally speaking, `shlex` is NOT meant to help prepare Windows commands ("outputs"). According to its documentation, it's meant as a convenience to help parsing _inputs_ to the program, like a simple configuration file or some parameters.

Longer `shlex` example and story in https://github.com/zephyrproject-rtos/west/commit/94f8a04ac83467b008f1a2aa5c4b437981367cf9